File: apps/api/src/AiEncyclopedia.Domain/Entities/Article.cs
using AiEncyclopedia.Domain.Enums;

namespace AiEncyclopedia.Domain.Entities;

public class Article
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TopicId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public string? Summary { get; set; }
    public ArticleStatus Status { get; set; } = ArticleStatus.Draft;
    public int ReadingTimeMinutes { get; set; }
    public long ViewCount { get; set; }
    public bool AiGenerated { get; set; }
    public AIProviderName? AiProvider { get; set; }
    public string? SeoTitle { get; set; }
    public string? SeoDescription { get; set; }
    public string[]? SeoKeywords { get; set; }
    public DateTimeOffset? PublishedAt { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;

    public Topic Topic { get; set; } = null!;
    public ICollection<ArticleTag> ArticleTags { get; set; } = new List<ArticleTag>();
    public ICollection<PdfDocument> PdfDocuments { get; set; } = new List<PdfDocument>();
}

File: packages/shared/src/types/article.ts
export type ArticleStatus = 'draft' | 'review' | 'published';
export type DifficultyLevel = 'beginner' | 'intermediate' | 'advanced';
export type AIProviderName = 'openai' | 'claude' | 'gemini';

export interface Category {
  id: string;
  name: string;
  slug: string;
  description?: string;
  parentId?: string;
  iconUrl?: string;
  articleCount: number;
}

export interface Topic {
  id: string;
  title: string;
  slug: string;
  categoryId: string;
  category?: Category;
  summary?: string;
  difficultyLevel: DifficultyLevel;
  articleCount: number;
}

export interface Tag {
  id: string;
  name: string;
  slug: string;
  color?: string;
}

export interface Article {
  id: string;
  topicId: string;
  topic?: Topic;
  title: string;
  slug: string;
  content: string;
  summary?: string;
  status: ArticleStatus;
  readingTimeMinutes: number;
  viewCount: number;
  aiGenerated: boolean;
  aiProvider?: AIProviderName;
  seoTitle?: string;
  seoDescription?: string;
  seoKeywords?: string[];
  tags: Tag[];
  publishedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface ArticleSummary {
  id: string;
  title: string;
  slug: string;
  summary?: string;
  readingTimeMinutes: number;
  viewCount: number;
  tags: Tag[];
  topic?: Pick<Topic, 'id' | 'title' | 'slug'>;
  publishedAt?: string;
}

export interface SearchResult {
  articles: ArticleSummary[];
  topics: Pick<Topic, 'id' | 'title' | 'slug' | 'summary'>[];
  totalCount: number;
}
